Puppet/Interpreter/Types.hs

```diff
@@ -187,12 +188,19 @@ data HieraQueryType
     , _mergeHashArray :: Bool
     } deriving (Show)
 
--- | The type of the Hiera API function.
-type HieraQueryFunc m = Container Text -- ^ All the variables that Hiera can interpolate, the top level ones being prefixed with ::
+-- | The type of the Hiera API function associated to one hiera file.
+type HieraQueryFunc m = Container Text -- ^ Scope: all variables that Hiera can interpolate (the top level ones are prefixed with ::)
                      -> Text -- ^ The query
                      -> HieraQueryType
                      -> m (S.Either PrettyError (Maybe PValue))
 
+-- | All available queries including the global and module layer
+-- The environment layer is not implemented
+type HieraQueryLayers m =
```

Comment:
Why not creating a dedicated type for this? That would make the code in `queryHiera` a tiny bit easier to read (I got stumped when I saw the `_1`).